src/methods/photos.js

```diff
@@ -145,6 +146,16 @@ export default function photos(): Object {
         url,
         method: "DELETE"
       });
+    },
+
+    downloadPhoto: (photo) => {
+      const urlComponents = getUrlComponents(photo["links"]["download_location"]);
```

Comment:
@samijaber can you double check my work here that based on doing this:

```js
unsplash.photos.getPhoto("mtNweauBsMQ")
  .then(toJson)
  .then(json => {
    unsplash.photos.downloadPhoto(json);
  });

// or if working with an array of photos
unsplash.search.photos("dogs", 1)
  .then(toJson)
  .then(json => {
    unsplash.photos.downloadPhoto(json["results"][0]);
  });
```

That this is how you'd access it...? I've been staring at Ruby and PHP too long to remember. 😬

Reply:
Yeah, that's fine (you can also use dot notation e.g. `photo.links.download_location` which is what we've set our eslint preference to be in other repos). 

More importantly, this function currently throws a `TypeError` if it receives a malformed object. We could either use lodash's `get` or a home-made version (which might be too much of a hassle), and log/throw a more descriptive error if we're not receiving the object we expect:

```js
{
  const downloadLocation = get(photo, 'links.download_location', undefined);
  if (downloadLocation === undefined) {
    throw new Error(`Object received is not a photo. ${photo}`);
  }
  // make request
  // ...
}
```

What do you think? I'd rather handle this here than have clients worry about input sanitization.

PS: Make sure to import `lodash.get` and not the entire module If you're opting for that route.

Reply:
Yup agree. I'll try importing just `lodash.get` and add the test case. Thanks!